drivers/net/ppp.c

```diff
@@ -747,21 +747,7 @@ static int ppp_send(const struct device *dev, struct net_pkt *pkt)
 			protocol = htons(PPP_IP);
 		} else if (net_pkt_family(pkt) == AF_INET6) {
 			protocol = htons(PPP_IPV6);
-		} else if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) &&
-			   net_pkt_family(pkt) == AF_PACKET) {
-			char type = (NET_IPV6_HDR(pkt)->vtc & 0xf0);
-
-			switch (type) {
-			case 0x60:
-				protocol = htons(PPP_IPV6);
-				break;
-			case 0x40:
-				protocol = htons(PPP_IP);
-				break;
-			default:
-				return -EPROTONOSUPPORT;
-			}
-		} else {
```

Comment:
Hmm, I am not sure how this is going to work. I mean that now for packet sockets, the protocol field is not set at all in this function, have you really tested that this works as expected?
Beside the `PPP_IPV6` != `NET_ETH_PTYPE_IPV6` so the protocol value would be set to incorrect value anyway.

Reply:
pkt family is now set already in net_context level and this code should be a dead code branch for 
AF_PACKET/SOCK_RAW/IPPROTO_RAW type sockets usage (to which type this code was initially added) and I have tested it to be working with that type.
But yeah, for another than IPPROTO_RAW type packets sockets, this ain't working, but it was assumed that there isn't that usage. Is that too much assumed? I can rollback this change for this file if wanted.